torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -523,23 +599,25 @@ struct GraphFuser {
     Node * producer_for_chunk_node = producer_for_chunk->node();
     JIT_ASSERT(producer_for_chunk_node->outputs().size() == 1);
 
-    // First, we'll add explicit broadcasts where necessary to make the chunk
-    // valid. Let's say we have:
-    // %z = aten::mul(%x, %y)
-    // %z.1, %z.2 = aten::chunk(%z, ...)
-    // ... = prim::FusionGroup(%z.1, %z.2, ...)
-    // It's possible that %x and %y do not have the same size as %z and
-    // need to be expanded first so that they can be chunked like %z
-    insertExplicitBroadcast(producer_for_chunk_node);
+    // Convert chunk to bchunk, if it isn't one already. The bchunk represents a
+    // broadcast and one or more chunk operations.
+    auto * bchunk = chunk;
+    if (chunk->kind() == prim::ConstantChunk) {
+      bchunk = promoteChunkToBroadcastingChunk(chunk);
+    }
+    size_t nchunks = bchunk->i(attr::chunks);
+    WithInsertPoint guard(bchunk);
 
-    // Make sure we lay out the nodes in the correct topological order.
-    // TODO: There should be some more enshrined way to do this
-    Node * insertion_point = chunk;
+    std::vector<Value*> producer_chunk_outputs;
+    for (size_t i = 0; i < nchunks; i++) {
+      producer_chunk_outputs.push_back(bchunk->output(4 * producer_index + i));
```

Comment:
What does this hardcoded `4` correspond to? Shouldn't this be `nchunks`?

Reply:
Yes, thank you for the catch!